File: lab9/CurrencyStorage.java
import java.util.*;

class CurrencyStorage<T extends Currency> implements IInterface<T> {
    private final List<T> list = new ArrayList<>();

    @Override
    public void add(T item) {
        list.add(item);
    }

    @Override
    public void printAll() {
        for (T item : list) {
            item.print();
        }
    }
}

File: lab9/CurrencyStorage.cs
using System.Collections.Generic;

class CurrencyStorage<T> : IInterface<T> where T : Currency
{
    private List<T> list = new List<T>();

    public void Add(T item)
    {
        list.Add(item);
    }

    public void PrintAll()
    {
        foreach (var item in list)
            item.Print();
    }
}